### PR TITLE
Complex test for PeerManager, and adding broadcasting metrics

### PR DESCRIPTION
**Pull request by mm-near**
Adding a complex test to PeerManager - to verify our AnnounceAccount behaviour under load.

This also requires adding a new metric (with broadcasting stats) to PeerManager.

### CODE REVIEW

File: integration-tests/src/tests/network/peer_manager.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                    let foo = successes.clone();
+                    pm.send(GetMetrics {}).then(move |res| {
+                        let metrics = res.unwrap();
+                        let sync_routing_table_sent = metrics
+                            .broadcast_messages
+                            .get_metric_with_label_values(&["SyncRoutingTable"])
+                            .unwrap()
+                            .get();
+                        if sync_routing_table_sent == 1 {
+                            // Try it couple times - we want to be sure that the didn't hit in the middle of processing.
```
Comment (nikurt) [inline integration-tests/src/tests/network/peer_manager.rs:144]: typo in `that the didn't hit`

File: integration-tests/src/tests/network/peer_manager.rs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        Arc<dyn Fn(&NetworkViewClientMessages) -> NetworkViewClientResponses + Send + Sync>,
+    >,
+) -> Addr<MockClientActor> {
+    SyncArbiter::start(5, move || MockClientActor {
+        on_view_client_callback: on_view_client_callback.clone(),
+    })
+}
+
+// This test is trying to see how PeerManager reacts to repeated AnnounceAccounts requests.
+// We want to make sure that we broadcast them only if necessary.
```
Comment (nikurt) [inline integration-tests/src/tests/network/peer_manager.rs:76]: Please consider clarifying what exactly "if necessary" means.

**Review by nikurt — APPROVED** [review]

File: integration-tests/src/tests/network/peer_manager.rs (hunk 1 above)
Comment (mm-near) [reply integration-tests/src/tests/network/peer_manager.rs]: done

File: integration-tests/src/tests/network/peer_manager.rs (hunk 2 above)
Comment (mm-near) [reply integration-tests/src/tests/network/peer_manager.rs]: done

**Pull request merged**